### DISCUSSION TITLE
Easy way to ignore a jsonkey?

### DISCUSSION THREAD

**Discussion by peterhijma** [simolus3/drift#1328] [Q&A]
See this issue: https://github.com/simolus3/moor/issues/952

Would I need to write a custom row class including all the methods like `copyWith`? Or is there an easier way? A hack is no problem. 

Previously I came away with adding the "nonsense" name as a key. Right now I get this error using that method:

```
flutter: Exception: type 'Null' is not a subtype of type 'int' in type cast
flutter: #0      _DefaultValueSerializer.fromJson (package:moor/src/runtime/data_class.dart:208:19)
```

Any help is appreciated.

**Comment by simolus3** [comment]
No, unfortunately this is not possible, you'll probably have to write your own data class.

Personally, I think that moor probably shouldn't be in the business of json serialization at all. There are other packages specialized on it, and moor has its own complicated problems to solve. Now that custom data classes are a thing, I don't see a compelling reason to further complicate the generation logic for what's not a core concern for moor.
